docs/ops/doc/ParameterConversion.rst

```diff
@@ -0,0 +1,141 @@
+	 */
+	public static RandomAccessibleInterval<DoubleType> convolveNaive(
+			final RandomAccessibleInterval<DoubleType> input,
+			final RandomAccessibleInterval<DoubleType> kernel
+        ) {
+            // convolve convolve convolve //
+        }
+
+This Op might work well, however if users have a small kernel that is *only* used for this Op, they may find it frustrating to represent that data as a ``RandomAccessibleInterval``. Fortunately, SciJava Ops allows us to write ``engine.convert`` Ops, so users can pass their data in a data structure more convenient for them.
+
+
+.. code-block:: java
+
+    Img<DoubleType> in = ...
+    // 3x3 averaging kernel
+    double m = 1/9;
+    double[] data = new double[] { //
+        m, m, m, //
+        m, m, m, //
+        m, m, m //
+    };
+    // Not ideal
+    Img<DoubleType> kernel = ArrayImgs.doubles(data, new long[] {3, 3});
+
+    var result = ops.op("filter.convolve").input(in, kernel).apply();
+
+In this case, users might find it nicer to specify their kernel as a ``double[][]``, which is much easier for users to construct
+
+.. code-block:: java
+
+    Img<DoubleType> in = ...
+    // 3x3 averaging kernel
+    double m = 1/9;
+    double[] kernel = new double[][] { //
+        new double[] { m, m, m}, //
+        new double[] { m, m, m}, //
+        new double[] { m, m, m} //
+    }
+
+    // Ideal case - no need to wrap to Img
+    var result = ops.op("filter.convolve").input(in, kernel).apply();
+
+The only step for us as the developer is to tell SciJava Ops that it can convert ``double[][]``\ s to ``RandomAccessibleInterval<DoubleType>``\ s, which we do with ``engine.convert`` Ops.
+
+A simple ``engine.convert`` Op
+==============================
+
+All ``engine.convert`` Ops are simple ``Function``\ s, that take as input the user argument to the Op, and return a *translation* of that data into the type expected by Ops. In our case, we want to convert *from* the user's ``double[][]`` into a ``RandomAccessibleInterval<DoubleType>``:
+
+.. code-block:: java
+
+    /**
+     * @param input the input data
+     * @return an output image whose values are equivalent to {@code input}s
+     *         values but whose element types are {@link BitType}s.
+     * @implNote op names='engine.convert', type=Function
+     */
+    public static RandomAccessibleInterval<DoubleType> arrayToDoubles(final double[][] input)
+    {
+        // Creates an empty image of doubles
+        var img = ArrayImgs.doubles(input.length, input[0].length);
+        var ra = img.randomAccess();
+        // Deep copies the double[][] into the RAI
+        for(int i = 0; i < input.length; i++) {
+            for(int j = 0; j < input[0].length; j++) {
+                ra.setPositionAndGet(i, j).set(input[i][j]);
+            }
+        }
+        return img;
+    }
+
+This Op, discovered through the SciJava Ops Indexer, is **all** that is needed to make the execution pattern we want functional.
```

Comment:
Bolding the word "all", and then later saying well actually, you might also need an `engine.copy` Op if you want to use computers, seems disingenuous to me.